tile_engine/ops/gemm/benchmark_gemm.hpp

```diff
@@ -0,0 +1,130 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2025, Advanced Micro Devices, Inc. All rights reserved.
+
+#pragma once
+
+#include <iostream>
+#include <vector>
+#include <filesystem>
+#include <memory>
+
+#include "ck/version.h"
+#include "ck/host_utility/device_prop.hpp"
+
+class Profiler
+{
+    public:
+    static Profiler& instance()
+    {
+        static Profiler instance;
+        return instance;
+    }
+
+    static std::string get_rocm_version()
+    {
+        std::ifstream version_file("/opt/rocm/.info/version");
+        if(version_file.is_open())
+        {
+            std::string version;
+            std::getline(version_file, version);
+            return version;
+        }
+        return "Unknown";
+    }
+
+    template <typename Kernel>
+    void benchmark_kernel(ck_tile::DeviceMem& c_m_n_dev_buf,
+                          ck_tile::HostTensor<CDataType>& c_m_n_host_result,
+                          ck_tile::HostTensor<CDataType>& c_m_n_dev_result,
+                          int verify,
+                          ck_tile::GemmHostArgs& args,
+                          const ck_tile::stream_config& s)
+    {
+        std::string description = Kernel::get_name();
+
+        GemmProblem problem{args.k_batch,
+                            args.M,
+                            args.N,
+                            args.K,
+                            args.stride_A,
+                            args.stride_B,
+                            args.stride_C,
+                            DataTypeTraits<ADataType>::name,
+                            DataTypeTraits<BDataType>::name,
+                            DataTypeTraits<AccDataType>::name,
+                            DataTypeTraits<CDataType>::name,
+                            ALayout::name,
+                            BLayout::name,
+                            CLayout::name};
+
+        KernelInstance kernel_instance{description, problem, {-1.0f, -1.0f, -1.0f}};
+
+        float avg_time = Kernel::launch(args, s);
```

Comment:
You should first check if the provided args are supported with `IsSupportedArgument` call https://github.com/ROCm/composable_kernel/blob/d3843d0ac0d5b32e67cece74e8baa707ee7b5fef/include/ck_tile/ops/gemm/kernel/gemm_kernel.hpp#L229